exercises/robot-name/bonus_test.go

```diff
@@ -5,25 +5,24 @@ package robotname
 import "testing"
 
 func TestCollisions(t *testing.T) {
-	m := map[string]bool{}
 	// Test uniqueness for new robots.
-	// 10k should be plenty to catch names generated
-	// randomly without a uniqueness check.
-	for i := 0; i < 10000; i++ {
-		n := New().Name()
-		if m[n] {
-			t.Fatalf("Name %s reissued after %d robots.", n, i)
-		}
-		m[n] = true
+	for i := len(seen); i <= maxNames-600000; i++ {
+		_ = New().getName(t, false)
 	}
+
 	// Test that names aren't recycled either.
 	r := New()
-	for i := 0; i < 10000; i++ {
+	for i := len(seen); i < maxNames; i++ {
```

Comment:
With the loop using maxNames to do a full uniqueness check, doing
```
go test -tags bonus -run TestCollisions
```
doesn't complete.  I gave up waiting after a couple minutes.
So I suppose the bonus_example.go algorithm isn't as efficient as it needs to be to complete with these changes. The bonus_example.go used with the test changes must be able to complete and pass the tests.

Have you found an algorithm with a reasonable bound on the runtime when doing a full uniqueness test for the bonus?  Or your suggestion to "limit the time spent randomly searching" could be used.

Reply:
FYI I'm about to dig up my solution to see how long it took--if I remember correctly it was fairly reasonable.